expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java

```diff
@@ -233,10 +173,108 @@ public String toString() {
         return toJsonString();
     }
 
+    private static JsonNode addFields(String json, Publication publication) {
+        var sortedJson = strToJsonWithSortedContributors(json);
+        injectHasFileEnum(publication, (ObjectNode) sortedJson);
+        expandLicenses(sortedJson);
+        injectAnthologyRelation(publication, (ObjectNode) sortedJson);
+        return sortedJson;
+    }
+
+    private static void injectAnthologyRelation(Publication publication, ObjectNode sortedJson) {
+        Optional.ofNullable(publication.getEntityDescription())
+                .flatMap(description -> Optional.ofNullable(description.getReference()))
+                .ifPresent(reference -> {
+                    var instanceType = reference.getPublicationInstance();
+                    var publicationContext = reference.getPublicationContext();
+
+                    if (instanceType instanceof BookAnthology) {
+                        addJoinField(sortedJson, "hasParts", null);
+                    } else if (isPartOfAnthology(publicationContext, instanceType)) {
+                        var parentId = ((Anthology) publicationContext).getId();
+                        var parentIdentifier = SortableIdentifier.fromUri(parentId).toString();
```

Comment:
We should probably move this check to the method above:

```
if (nonNull(parent)) {
            newNode.put("parent", parent);
        }
```
        
Also do we want to inject joinField without parentId when parentId is missing, like:
```
 "joinField" : {
    "name" : "partOf"
  }
```

Reply:
Thanks! Added a test case and logic to handle this type of invalid data now. Better now, or do we need more tests?

We have to add the joinField with a dummy parent ID because OpenSearch requires a parent ID for child documents (cannot be null or missing). It does not matter what the dummy ID is, and a query can be made to find all documents referencing a parent that does not exist in the index.